src/component/logs.rs

```diff
@@ -0,0 +1,104 @@
+use std::fs::read_to_string;
+use std::mem::replace;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+use tempfile::NamedTempFile;
+
+use crate::error::WorkerError;
+use subprocess::{PopenConfig, Redirection};
+
+static DEDUP_COUNTER: AtomicU64 = AtomicU64::new(0);
+
+#[derive(Debug)]
+pub struct LogTracker {
+    // Tracks when a different log tracker is switched to
+    dedup_number: u64,
+    policy_handle: Arc<LogPolicy>,
+}
+
+impl LogTracker {
+    pub fn new() -> Self {
+        Self {
+            dedup_number: DEDUP_COUNTER.fetch_add(1, Ordering::SeqCst),
+            policy_handle: LogPolicy::new_no_op_policy(),
+        }
+    }
+
+    pub fn create_associated_policy(&mut self) -> Result<Arc<LogPolicy>, WorkerError> {
```

Comment:
So does this just set the policy to be "write to a file"?

Reply:
The bit that's doing the logging just gets a log policy. An associated policy is a file writing policy that the `LogTracker` knows about, and can use to service requests to `logs`